apps/zui/scripts/merge-mac-release-files.mjs

```diff
@@ -0,0 +1,226 @@
+    return;
+  } else {
+    console.log('Release found')
+  }
+
+  const localLatestMacYmlExists = fs.existsSync(LOCAL_FILE_PATH);
+
+  if (!localLatestMacYmlExists) {
+    console.log(`[local] could not find ${FILE_NAME}. Skipping merge`);
+    return;
+  } else {
+    console.log(`[local] ${FILE_NAME} found`);
+  }
+
+  const localLatestMacYmlContent = fs.readFileSync(LOCAL_FILE_PATH, { encoding: 'utf8' });
+
+  const localPlatform = getPlatformFromLatestMacYml(localLatestMacYmlContent);
+
+  if (localPlatform === 'none' || localPlatform === 'both') {
+    console.log(`[local] ${FILE_NAME} invalid. Platform: ${localPlatform}. Skipping merge`)
+    return;
+  } else {
+    console.log(`[local] ${FILE_NAME} valid: Platform: ${localPlatform}`)
+  }
+
+  const localPlatformPresentRemotely = currentRelease.assets.find(asset => {
+    return asset.name === `latest-mac-${localPlatform}.yml`;
+  });
+
+  if (localPlatformPresentRemotely) {
+    try {
+      await client.request(`DELETE ${URL}/assets/${localPlatformPresentRemotely.id}`);
+      console.log(`[remote] deleted latest-mac-${localPlatform}.yml`)
+    } catch(e) {
+      console.log(`[remote] error deleting latest-mac-${localPlatform}.yml. Skipping merge`)
+      console.log(e);
+      return;
+    }
+  }
+
+  const uploadUrl = currentRelease.upload_url;
+  const localAssetStream = new Readable();
+  localAssetStream.push(localLatestMacYmlContent);
+  localAssetStream.push(null);
+
+  try {
+    await client.rest.repos.uploadReleaseAsset({
+      url: uploadUrl,
+      headers: {
+        'content-type': 'application/octet-stream',
+        'content-length': Buffer.byteLength(localLatestMacYmlContent),
+      },
+      name: `latest-mac-${localPlatform}.yml`,
+      data: localAssetStream,
+    })
+    console.log(`[remote] latest-mac-${localPlatform}.yml uploaded`)
+  } catch(e) {
+    console.log(`[remote] error uploading latest-mac-${localPlatform}.yml. Skipping merge`);
+    console.log(e);
+    return;
+  }
+
+  const remotePlatform = localPlatform === 'intel' ? 'arm' : 'intel';
+
+  const remotePlatformFileExists = currentRelease.assets.find(asset => {
+    return asset.name === `latest-mac-${remotePlatform}.yml`;
+  })
+
+  if (!remotePlatformFileExists) {
+    console.log(`[remote] latest-mac-${remotePlatform}.yml does not exist. Skipping merge`)
+    return;
+  } else {
+    console.log(`[remote] latest-mac-${remotePlatform}.yml found`)
+  }
+
+  let remotePlatformFile = null;
+
+  try {
+    remotePlatformFile = await client.request(`GET ${URL}/assets/${remotePlatformFileExists.id}`, {
+      headers: {
+        accept: 'application/octet-stream'
+      }
+    });
+    console.log(`[remote] latest-mac-${remotePlatform}.yml downloaded`)
+  } catch(e) {
+    console.log(`[remote] error downloading latest-mac-${remotePlatform}.yml. Skipping merge`);
+    console.log(e);
+    return;
+  }
+
+  const remoteLatestMacYmlContent = new TextDecoder().decode(remotePlatformFile.data);
```

Comment:
Pretty sure remotePlatformFile.data is a Buffer so you should be able to do the following and not need the TextDecoder import:
```suggestion
  const remoteLatestMacYmlContent =remotePlatformFile.data.toString();
```

Reply:
@mattnibs: I tested out the proposed change and it didn't work as is. I looked closer and found that the `.data` in the GitHub response is an `ArrayBuffer`, not a `Buffer`. I'm not super familiar with this but I did some reading/searching about best practices for converting ArrayBuffer to strings and the [first StackOverflow post I found](https://stackoverflow.com/questions/6965107/converting-between-strings-and-arraybuffers) has a top-voted answer of folks recommending `TextDecoder`. Therefore I'm inclined to leave this as is for now.